trace_decoder/src/compact/compact_prestate_processing.rs

```diff
@@ -1453,6 +1466,24 @@ mod tests {
         hex::decode(b_str).unwrap()
     }
 
+    // TODO: Move behind a feature flag just used for debugging (but probably not
+    // `debug`)...
+    fn parse_just_to_instructions(bytes: Vec<u8>) -> CompactParsingResult<Vec<Instruction>> {
+        let witness_bytes = WitnessBytes::<DebugCompactCursor>::new(bytes);
+        let (_, entries) = witness_bytes.process_into_instructions_and_header()?;
+
+        Ok(entries
+            .intern
+            .into_iter()
+            .map(|entry| match entry {
+                WitnessEntry::Instruction(instr) => instr,
+                _ => unreachable!(
+                    "Found a non-instruction at a stage when we should only have instructions!"
```

Comment:
We're probably better handling this nicely over panicking. We can return a proper error here instead.